### PR TITLE
feat(proxy-host-header): add proxy host header INTER-346

### PR DESCRIPTION
**Pull request by Orkuncakilkaya**

### CODE REVIEW

**Comment by ilfa** [comment]
Can we remove suffix list too?

**Review by necipallef — DISMISSED** [review]

**Comment by necipallef** [comment]
We can remove `src/utils/domain-suffix-list-reversed.json` file and `punycode` dependency. We can do it in this PR or create a separate PR, I am fine with both

**Comment by Orkuncakilkaya** [comment]
I removed punycode & domain suffix list

File: test/utils/addProxyIntegrationHeaders.test.ts [hunk 1]
```diff
@@ -15,21 +15,21 @@ describe('addProxyIntegrationHeaders', () => {
     }
   })
   test('when PROXY_SECRET is set', () => {
-    addProxyIntegrationHeaders(headers, env)
+    addProxyIntegrationHeaders(headers, 'https://example.com/worker/result', env)
```
Comment (ilfa) [inline test/utils/addProxyIntegrationHeaders.test.ts:18]: We need to check `FPJS-Proxy-Forwarded-Host` header too

**Review by ilfa — CHANGES_REQUESTED** [review]

**Review by necipallef — DISMISSED** [review]

File: test/utils/addProxyIntegrationHeaders.test.ts (hunk 1 above)
Comment (Orkuncakilkaya) [reply test/utils/addProxyIntegrationHeaders.test.ts:18]: Just pushed a commit for this

File: test/utils/addProxyIntegrationHeaders.test.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ describe('addProxyIntegrationHeaders', () => {
     addProxyIntegrationHeaders(headers, 'https://example.com/worker/result', env)
     expect(headers.get('FPJS-Proxy-Secret')).toBe('secret_value')
     expect(headers.get('FPJS-Proxy-Client-IP')).toBe('19.117.63.126')
+    expect(headers.get('FPJS-Proxy-Forwarded-Host')).toBe('example.com')
```
Comment (ilfa) [inline test/utils/addProxyIntegrationHeaders.test.ts:21]: Let's add it for the next 2 tests too

**Review by sshelomentsev — APPROVED** [review]

**Review by necipallef — APPROVED** [review]

**Review by ilfa — APPROVED** [review]

**Pull request merged**